feat: add `number/float32/base/ulp-difference`

anandkaranubc opened the pull request:
Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the implementation for `number/float32/base/ulp-difference`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolve none.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: temp commit

anandkaranubc committed:
feat: temp commit

anandkaranubc committed:
Merge branch 'develop' into feat/ulp-diff-f32

anandkaranubc committed:
feat: add `number/float32/base/ulp-difference`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/number/float32/base/ulp-difference/lib/main.js:
```diff
@@ -0,0 +1,104 @@
+	wx = monotoneKey( wx );
+	wy = monotoneKey( wy );
+
+	// Return a double as a result, which can exactly represent the ULP difference for all representable single-precision floating-point numbers:
```
Fine to return a double here as well?

kgryte replied:
Yes, that makes sense to me.

anandkaranubc replied:
Also, I hope the complete implementation makes sense here 😄

anandkaranubc committed:
docs: add comment

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: fix typo in comment

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented:
Maybe good to carry this small change to `float64` too:

https://github.com/stdlib-js/stdlib/pull/7451/commits/2adc11c6722207f19d45d1ce92b95f00a357903d

kgryte commented:
Yes, please!

kgryte commented on lib/node_modules/@stdlib/number/float32/base/ulp-difference/package.json:
```diff
@@ -0,0 +1,68 @@
+    "numbers",
+    "ulp",
+    "float32",
+    "double"
```
```suggestion
    "float"
```

kgryte committed:
chore: update keywords

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/number/float32/base/ulp-difference/test/test.js:
```diff
@@ -0,0 +1,112 @@
+	t.end();
+});
+
+tape( 'the function returns the ULP difference between two double-precision floating-point numbers', function test( t ) {
```
```suggestion
tape( 'the function returns the ULP difference between two single-precision floating-point numbers', function test( t ) {
```

kgryte committed:
test: fix description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
